internal/provider/resource_gitlab_managed_license_test.go

```diff
@@ -144,3 +180,25 @@ resource "gitlab_managed_license" "fixme" {
 }
 	`, rInt, status)
 }
+
+// This method is used as a skipfunc to ensure we're both running in EE, and the EE
+// version is less than 15, since existing values were deprecated in v15.
+func checkCEAndDeprecatedVersion() (bool, error) {
```

Comment:
We have a generic function similar to this already:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/49db985b79f667da9244a57d9ff4f107f337b506/internal/provider/util.go#L227

Could you do something like this for the `SkipFunc` instead?

```golang
SkipFunc: orSkipFuncs(isRunningInCE, isGitLabVersionLessThan(testGitlabClient, "15.0")
```

(`orSkipFuncs` doesn't exist; you'd need to write that)

Reply:
@armsnyder  - For sure. Which of these do you think is better?

`func orSkipFun(first func() (bool, error), second func() (bool, error)) func() (bool error)`
or 
`orSkipFunc(functions []func() (bool, error)) func() (bool, error)`

The first one is going to result in code within the func itself that's a lot more readable, and a skipFunc declaration that matches what your review finding is suggesting, but it would only ever accept 2 functions so if we wanted to do a 3-deep "or" chain we'd need to chain the "orSkipFunc" method twice. The second option would allow for many skipFuncs, but the skipFunc declaration in the test itself would need to be something like:

```
SkipFunc: orSkipFunc([]func() (bool, error){
	isRunningInCE,
	isGitLabVersionLessThan(testGitlabClient, "15"),
}),
```

I'm leaning to the first option, just because we haven't needed an "OR" function to-date, so the likelihood of needing one 3-deep is pretty slim, so I favor the readability for now. Thoughts?

Reply:
What about something like this:

```go
func orSkipFunc(skipFuncs ...func() (bool, error)) (bool, error) {
	for _, skipFunc := range skipFuncs {
		skip, err := skipFunc()
		if err != nil {
			return false, err
		}
		if skip {
			return true, nil
		}
	}
	return false, nil
}
```

... then call it like this:

```go
SkipFunc: orSkipFunc(isRunningInCE, isGitLabVersionLessThan(testGitlabClient, "15"))
```

Reply:
Getting closer... the func needs to return a func though. Combining both of your suggestions...

```golang
func orSkipFunc(skipFuncs ...func() (bool, error)) func() (bool, error) {
	switch len(skipFuncs) {
	case 0:
		panic("zero arguments")
	case 1:
		return skipFuncs[0]
	case 2:
		return func() (bool, error) {
			v0, err := skipFuncs[0]()
			if err != nil {
				return false, err
			}
			v1, err := skipFuncs[1]()
			if err != nil {
				return false, err
			}
			return v0 || v1, nil
		}
	default:
		return orSkipFunc(skipFuncs[0], orSkipFunc(skipFuncs[1:]...))
	}
}
```

Reply:
Getting closer... the func needs to return a func though. Combining both of your suggestions...

```golang
type SkipFunc = func() (bool, error)

func orSkipFunc(a SkipFunc, b ...SkipFunc) SkipFunc {
	switch len(b) {
	case 0:
		return a
	case 1:
		return func() (bool, error) {
			va, err := a()
			if err != nil {
				return false, err
			}
			vb, err := b[0]()
			if err != nil {
				return false, err
			}
			return va || vb, nil
		}
	default:
		return orSkipFunc(a, orSkipFunc(b[0], b[1:]...))
	}
}
```

Reply:
Perfect, I actually implemented the "only 2" option in the last commit that has almost explicitly what you have in "case 1" implemented, but I'll update the approach. I really like defining a specific SkipFunc type, makes the code _way_ easier to read.

Just out of curiosity, is there a reason to use recursion as opposed to just using range to loop through the SkipFunc and short-circuit returning on the first "true" or error value?

Reply:
I'd say personal preference and background :)

Reply:
Fair enough. I tend to prefer avoiding recursion, so I committed the one I'd written above. Feel free to leave comments if we don't like it 😄